README.md

```diff
@@ -48,9 +48,11 @@ Additionally, the following custom functions are supported:
 - `fromConfigMap` returns the value of a key inside a `ConfigMap`. For example,
   `{{ fromConfigMap "namespace" "config-map-name" "key" }}`.
 - `fromSecret` returns the value of a key inside a `Secret`. For example,
-  `{{ fromSecret "namespace" "secret-name" "key" }}`.
+  `{{ fromSecret "namespace" "secret-name" "key" }}`. If the `protect` template function is enabled,
```

Comment:
I'm not clear how the protect function could be disabled.  If it can be disabled through an error I would not want customer data to miss being encrypted though.

Reply:
It won't be enabled if the `EncryptionMode` is set to `EncryptionDisabled` or `DecryptionEnabled`.

When performing Hub policy templates, it will be set to `EncryptionEnabled`, which means `fromSecret` will always return encrypted data and if it fails to do so for an unexpected reason, template processing would just halt and the policy would become non-compliant. The `protect` template function is also available for more advanced use-cases.

When performing managed cluster templates, the `EncryptionMode` will be set to `DecryptionEnabled`, which means that `fromSecret` will continue to work as it does today and that the `protect` template function will not be available. If a user tries to use it, it'll say that the `protect` template function is not defined.

In a follow-up PR, the `DecryptionEnabled` mode will indicate that encrypted strings should be searched for and decrypted before other template actions.